streams-contrib/streams-processor-regex/src/main/java/org/apache/streams/regex/RegexUtils.java

```diff
@@ -36,6 +36,7 @@
 
     private static final Map<String, Pattern> patternCache = Maps.newConcurrentMap();
     private final static Logger LOGGER = LoggerFactory.getLogger(RegexUtils.class);
+    private final static long REGEX_MATCH_THREAD_KEEP_ALIVE_DEFAULT_TIMEOUT = 10000L; //10 seconds
```

Comment:
This might be too long of a default.  If there are 5K documents in queue and in the worst case, each one hits the max you are looking at 500K seconds to complete.  IMO, if the regex can't extract in < 1/2 second, it is going to be a problem.